### PR TITLE
Add a new `lint` task to replace the `lint-go` task and use `golangci-lint`

### PR DESCRIPTION
**Pull request by autarch**
This gets rid of the ancient fork of `golint` we were using and replaces it with `golangci-lint`, fixing quite a few linting issues found by `golangci-lint`.

It also removes the `format-go` task, as that can be done with `precious` and `golangci-lint` as well.

### CODE REVIEW

**Comment by autarch** [comment]
Since this is a big change, I will not merge this until after our next release.

File: mongoimport/common.go
Comment (autarch) [reply mongoimport/common.go:143]: Every single caller set `quorum` to `2`, so I just made this a `const`.

File: mongoimport/csv.go
Comment (autarch) [reply mongoimport/csv.go:157]: I don't think so. This function is using named return values. I'm not a fan of this, but this diff is already huge, so I wanted to avoid optional changes.

File: common/txn/buffer.go
Comment (autarch) [reply common/txn/buffer.go:287]: Yes, good call! Done.

File: mongorestore/oplog.go
Comment (autarch) [reply mongorestore/oplog.go]: Done.

File: mongoimport/common.go
Comment (tdq45gj) [reply mongoimport/common.go:143]: It seems that the function's comment wasn't updated.

**Review by tdq45gj — APPROVED** [review]
LGTM % one remaining comment.

File: mongoimport/common.go
Comment (autarch) [reply mongoimport/common.go:141]: Updated.

File: etc/golines-wrapper.sh
Comment (edobranov) [inline etc/golines-wrapper.sh:1]: Might've missed it, but are we using this anywhere? (if not, do we think we'll use it?)

**Review by edobranov — APPROVED** [review]
LGTM % one question

File: etc/golines-wrapper.sh
Comment (autarch) [reply etc/golines-wrapper.sh:1]: Oops, we will use it in a follow-up PR. Not sure how that snuck into this one, but I'll leave it since the next PR is just waiting on this one to be merged.

**Pull request merged**
